Mutagen.Bethesda.Analyzers.Skyrim/Record/Location/CircularLocationAnalyzer.cs

```diff
@@ -0,0 +1,49 @@
+using Mutagen.Bethesda.Analyzers.SDK.Analyzers;
+using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Plugins.Cache;
+using Mutagen.Bethesda.Skyrim;
+
+namespace Mutagen.Bethesda.Analyzers.Skyrim.Record.Location;
+
+class CircularLocationAnalyzer : IContextualRecordAnalyzer<ILocationGetter>
+{
+    public static readonly TopicDefinition CircularLocation = MutagenTopicBuilder.DevelopmentTopic(
+            "Circular Location",
+            Severity.CTD)
+        .WithoutFormatting("Location is its own parent location - this leads to CTDs and infinite loading screens");
```

Comment:
The cause of an effect (CTD, infinite loading screens) is usually not documented in the description, but some other place such as a discussion entry [here](https://github.com/Mutagen-Modding/Mutagen.Bethesda.Analyzers/discussions). If you want, you can open a discussion for this, and also create an actual topic instead of a development topic (which we are planning to convert in the future anyway).

Reply:
Yup ^  agreed

The reasoning behind this is that the message here is what will get displayed in the "IDE".  These are usually succinct (but not always, lol)
![image](https://github.com/user-attachments/assets/c6b17fa2-396d-4a1a-be49-f27f97fdc6ba)

Full discussions/reasoning can be included elsewhere